tests/Integration/Behaviour/Features/Scenario/CartRule/set_mixed_cart_rule_restrictions.feature

```diff
@@ -0,0 +1,125 @@
+    And I add product "product1" with following information:
+      | name[en-US] | bottle of beer |
+      | type        | virtual        |
+    And I add product "product2" with following information:
+      | name[en-US] | T-shirt nr1 |
+      | type        | standard    |
+    And I add product "product3" with following information:
+      | name[en-US] | Shirt - Dom & Jquery |
+      | type        | standard             |
+    And I create cart rule "rule_free_shipping_1" with following properties:
+      | name[en-US]       | free shipping 1      |
+      | is_active         | true                 |
+      | allow_partial_use | false                |
+      | priority          | 1                    |
+      | valid_from        | 2022-01-01 11:00:00  |
+      | valid_to          | 3001-01-01 12:00:00  |
+      | total_quantity    | 10                   |
+      | quantity_per_user | 10                   |
+      | free_shipping     | true                 |
+      | code              | rule_free_shipping_1 |
+    And I create cart rule "rule_50_percent" with following properties:
+      | name[en-US]                  | Half the price         |
+      | is_active                    | true                   |
+      | allow_partial_use            | true                   |
+      | priority                     | 2                      |
+      | valid_from                   | 2022-01-01 11:00:00    |
+      | valid_to                     | 3001-01-01 12:00:00    |
+      | total_quantity               | 10                     |
+      | quantity_per_user            | 12                     |
+      | free_shipping                | false                  |
+      | code                         | rule_50_percent        |
+      | discount_percentage          | 50                     |
+      | apply_to_discounted_products | false                  |
+      | discount_application_type    | order_without_shipping |
+    And I create cart rule "rule_70_percent" with following properties:
+      | name[en-US]                  | Half the price         |
+      | is_active                    | true                   |
+      | allow_partial_use            | true                   |
+      | priority                     | 3                      |
+      | valid_from                   | 2022-01-01 11:00:00    |
+      | valid_to                     | 3001-01-01 12:00:00    |
+      | total_quantity               | 10                     |
+      | quantity_per_user            | 12                     |
+      | free_shipping                | false                  |
+      | code                         | rule_70_percent        |
+      | discount_percentage          | 70                     |
+      | apply_to_discounted_products | false                  |
+      | discount_application_type    | order_without_shipping |
+    And cart rule "rule_free_shipping_1" should have the following properties:
```

Comment:
Should you clean them in case some additional scenarios are added? Like you did in the product restrictions feature file

Reply:
```suggestion
    And I clear all product restrictions for cart rule rule_free_shipping_1
    And I save all the restrictions for cart rule rule_free_shipping_1
    And I clear all product restrictions for cart rule rule_50_percent
    And I save all the restrictions for cart rule rule_50_percent
    And I clear all product restrictions for cart rule rule_70_percent
    And I save all the restrictions for cart rule rule_70_percent
    And cart rule "rule_free_shipping_1" should have the following properties:
```